ext/storage_sources/s3_store/test/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+project(s3_store_tests CXX)    
+include(${CMAKE_SOURCE_DIR}/test/ctest_helpers.cmake)
+file(GLOB test_src "*.cpp")
```

Comment:
We should  avoid using `GLOB` in the build system, since this command won't be re-run if the CMakeLists doesn't change i.e. if we add a new unit test file it won't be picked up by CMake (further the official CMake docs explicitly do not recommended the use of `GLOB` for this purpose: https://cmake.org/cmake/help/latest/command/file.html#filesystem). 
I understanding you're trying to automate the definition of the tests, but its much better to be explicit here and manually define each `create_test_executable`. Its also better from a readability perspective to see all the test definitions laid out.

Reply:
I think we're moving away from multiple tests and placing all the unit tests within one .cpp source file so it shouldn't be a problem removing this. Are there any downsides to this new approach?

Reply:
No problem with that approach, its perfectly fine to have them all in a single file (main point was to just remove the use of `GLOB` and be explicit with the source files this CMakeLists depends on)